rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/rules/AnalyzeJSPFileRuleProvider.java

```diff
@@ -89,9 +93,9 @@ public void perform(GraphRewrite event, EvaluationContext context, JspSourceFile
             catch (Exception e)
             {
                 ClassificationService classificationService = new ClassificationService(event.getGraphContext());
-                classificationService.attachClassification(context, sourceFile, JspSourceFileModel.UNPARSEABLE_JSP_CLASSIFICATION,
-                            JspSourceFileModel.UNPARSEABLE_JSP_DESCRIPTION);
-                sourceFile.setParseError("JSP file could not be parsed: " + e.getMessage());
+                classificationService.attachClassification(context, sourceFile, UNPARSEABLE_JSP_CLASSIFICATION, UNPARSEABLE_JSP_DESCRIPTION);
+                if (! Boolean.TRUE.equals(sourceFile.isIgnoreParseError()))
```

Comment:
I'd prefer storing the parse error either way and then dealing with whether to display it at the presentation level.